src/main/java/net/mcreator/ui/dialogs/tools/WoodPackMakerTool.java

```diff
@@ -0,0 +1,447 @@
+		slabBlock.texture = planksTextureName;
+		slabBlock.textureTop = planksTextureName;
+		slabBlock.textureFront = planksTextureName;
+		slabBlock.soundOnStep = new StepSound(workspace, "WOOD");
+		slabBlock.hardness = 2 * factor;
+		slabBlock.resistance = 3 * factor;
+		slabBlock.breakHarvestLevel = 0;
+		slabBlock.flammability = (int) Math.round(5 * factor);
+		mcreator.getWorkspace().getModElementManager().storeModElementPicture(slabBlock);
+		mcreator.getWorkspace().addModElement(slabBlock.getModElement());
+		mcreator.getWorkspace().getGenerator().generateElement(slabBlock);
+		mcreator.getWorkspace().getModElementManager().storeModElement(slabBlock);
+
+		// we use Block GUI to get default values for the block element (kinda hacky!)
+		Block fenceBlock = (Block) ModElementTypeRegistry.REGISTRY.get(ModElementType.BLOCK)
+				.getModElement(mcreator, new ModElement(workspace, name + "Fence", ModElementType.BLOCK), false)
+				.getElementFromGUI();
+		fenceBlock.name = name + " Fence";
+		fenceBlock.blockBase = "Fence";
+		fenceBlock.texture = planksTextureName;
+		fenceBlock.soundOnStep = new StepSound(workspace, "WOOD");
+		fenceBlock.hardness = 2 * factor;
+		fenceBlock.resistance = 3 * factor;
+		fenceBlock.breakHarvestLevel = 0;
+		fenceBlock.flammability = (int) Math.round(5 * factor);
+		mcreator.getWorkspace().getModElementManager().storeModElementPicture(fenceBlock);
+		mcreator.getWorkspace().addModElement(fenceBlock.getModElement());
+		mcreator.getWorkspace().getGenerator().generateElement(fenceBlock);
+		mcreator.getWorkspace().getModElementManager().storeModElement(fenceBlock);
+
+		// we use Block GUI to get default values for the block element (kinda hacky!)
+		Block fenceGateBlock = (Block) ModElementTypeRegistry.REGISTRY.get(ModElementType.BLOCK)
+				.getModElement(mcreator, new ModElement(workspace, name + "FenceGate", ModElementType.BLOCK), false)
+				.getElementFromGUI();
+		fenceGateBlock.name = name + " Fence Gate";
+		fenceGateBlock.blockBase = "FenceGate";
+		fenceGateBlock.texture = planksTextureName;
+		fenceGateBlock.soundOnStep = new StepSound(workspace, "WOOD");
+		fenceGateBlock.hardness = 2 * factor;
+		fenceGateBlock.resistance = 3 * factor;
+		fenceGateBlock.breakHarvestLevel = 0;
+		fenceGateBlock.flammability = (int) Math.round(5 * factor);
+		mcreator.getWorkspace().getModElementManager().storeModElementPicture(fenceGateBlock);
+		mcreator.getWorkspace().addModElement(fenceGateBlock.getModElement());
+		mcreator.getWorkspace().getGenerator().generateElement(fenceGateBlock);
+		mcreator.getWorkspace().getModElementManager().storeModElement(fenceGateBlock);
+
+		// Tags
+		Tag woodBlockTag = (Tag) ModElementTypeRegistry.REGISTRY.get(ModElementType.TAG)
+				.getModElement(mcreator, new ModElement(workspace, name + "BlocksTag", ModElementType.TAG), false)
+				.getElementFromGUI();
+		woodBlockTag.namespace = "forge";
+		woodBlockTag.name = name.toLowerCase().replace(" ", "_") + "_log";

```

Comment:
Use RegistryNameFixer.fix(String) where you want to do things like this for a proper and tested solution.

Reply:
Ok, perfect!